### PR TITLE
Add BERT QA

### PR DESCRIPTION
**Pull request by pengshuang**
see issue https://github.com/allenai/allennlp-reading-comprehension/issues/10

@dirkgr This is my implementation of BERT-QA (the UT is coming soon), but I have some problems when compared with the official huggingface's version. The F1 and EM I got are lower in SQuAD 1.1 dataset. 

**F1-Score Comparison**
Huggingface: 88.0%
Ours: 82.1%

Interestingly,  the comparison performance in Chinese MRC tasks are consistent. I have a hard time to figure the problem (but failed) and thus hope to get your suggestions. 

Feel free to contact me if you have any ideas.

### CODE REVIEW

File: allennlp_rc/dataset_readers/bert_squad.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+                            tokens.append(token)
+
+                        # SEP token
+                        tokens.append(Token(sep_token))
+
+                        # Paragraph
+                        paragraph_start_id = len(tokens)
+                        for i in range(doc_span.length):
+                            split_token_index = doc_span.start + i
+                            token_to_orig_map[len(tokens)] = tok_to_orig_index[split_token_index]
```
Comment (dirkgr) [inline allennlp_rc/dataset_readers/bert_squad.py]: `token_to_orig_map` is not read anywhere, just written to.

Comment (dirkgr) [reply allennlp_rc/dataset_readers/bert_squad.py]: And I think `tok_to_orig_index` is only here to set up `token_to_orig_map`, so it's not needed either.

File: allennlp_rc/dataset_readers/bert_squad.py [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+
+                            is_max_context = _check_is_max_context(doc_spans, doc_span_index,
+                                                                   split_token_index)
+                            token_is_max_context[len(tokens)] = is_max_context
```
Comment (dirkgr) [inline allennlp_rc/dataset_readers/bert_squad.py]: `token_is_max_context` is also never read again.

File: allennlp_rc/dataset_readers/bert_squad.py [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+                        )
+
+                        yield instance
+
+    @overrides
+    def text_to_instance(self,
+                         qas_id: str,
+                         question_text: str,
+                         bert_tokens: List[Token],
+                         passage_tokens: List[Token],
```
Comment (dirkgr) [inline allennlp_rc/dataset_readers/bert_squad.py]: `passage_tokens` is unused.

**Comment by dirkgr** [comment]
This code also doesn't add a CLS token, but I guess that's not strictly necessary for this problem.

**Comment by pengshuang** [comment]
> This code also doesn't add a CLS token, but I guess that's not strictly necessary for this problem.

Actually, here we don't need to add a CLS token manually,  the **token_indexers** will automatically add the CLS and SEP tokens later.

File: allennlp_rc/dataset_readers/bert_squad.py (hunk 1 above)
Comment (pengshuang) [reply allennlp_rc/dataset_readers/bert_squad.py]: I have found the problem that accounts for the difference in score and will fix it ASAP.
In my fixed version, `token_to_orig_map ` is used.

File: allennlp_rc/dataset_readers/bert_squad.py (hunk 3 above)
Comment (pengshuang) [reply allennlp_rc/dataset_readers/bert_squad.py]: Thanks for your careful review, I will fix it.

**Comment by pengshuang** [comment]
I have found the reason that accounts for the difference in score and will fix it with UT in this weekend.

**Comment by dirkgr** [comment]
I'm closing this for lack of activity, and also because I already took the good bits and put them into #19. @pengshuang, even though we're not merging this, thanks for the contribution! I stole a few things from here.

**Pull request closed without merging**